File: client/src/bounds.ts
export default class Bounds {
    x1: number
    y1: number
    x2: number
    y2: number
    width: number
    height: number

    constructor(x1: number, y1: number, x2: number, y2: number) {
        this.x1 = x1
        this.y1 = y1
        this.x2 = x2
        this.y2 = y2
        this.width = this.x2 - this.x1
        this.height = this.y2 - this.y1
    }

    collision(other: Bounds): boolean {
        return !(other.x1 > this.x2 || other.x2 < this.x1 || other.y1 > this.y2 || other.y2 < this.y1)
    }

    recompute() {
        this.width = this.x2 - this.x1
        this.height = this.y2 - this.y1
    }
}

File: service/src/shooter/bounds.go
package main

type Bounds struct {
	X1 int64
	Y1 int64
	X2 int64
	Y2 int64
	Width int64
	Height int64
}

func NewBounds(x1, y1, x2, y2 int64) *Bounds {
	bounds := Bounds{
		X1: x1,
		Y1: y1,
		X2: x2,
		Y2: y2,
	}
	return &bounds
}

func (b *Bounds) Collision(other *Bounds) bool {
	return !(other.X1 > b.X2 || other.X2 < b.X1 || other.Y1 > b.Y2 || other.Y2 < b.Y1)
}

func (b *Bounds) recompute() {
	b.Width = b.X2 - b.X1
	b.Height = b.Y2 - b.Y1
}
